### PR TITLE
When creating a new account (via domain admin) it is possible to select “root admin” as the role for the new user

### PR DESCRIPTION
**Pull request by rafaelweingartner**
## Description

Domain admins should not be able to assign the role of root admin to new users. Therefore, the role ‘root admin’ (or any other of the same type) should not be visible to domain admin accounts.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?
I created unit test that cover all of the modified code. Moreover, I tested locally.

## Checklist:

- [X] I have read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/master/CONTRIBUTING.md) document.
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
Testing
- [X] I have added tests to cover my changes.
- [X] All relevant new and existing integration tests have passed.
- [ ] A full integration testsuite with all test that can run on my environment has passed.

### CODE REVIEW

File: api/src/main/java/org/apache/cloudstack/acl/RoleService.java [hunk 1]
```diff
@@ -17,38 +17,62 @@
...
-    Role findRole(final Long id);
-    Role createRole(final String name, final RoleType roleType, final String description);
-    Role updateRole(final Role role, final String name, final RoleType roleType, final String description);
-    boolean deleteRole(final Role role);
 
-    RolePermission findRolePermission(final Long id);
-    RolePermission findRolePermissionByUuid(final String uuid);
+    /**
+     *  Searches for a role with the given ID. If the ID is null or less than zero, this method will return null.
+     *  This method will also return null if no role is found with the provided ID. Moreover, we will check if the requested role is of 'Admin' type; roles with 'Admin' type should only be visible to 'root admins'.  Therefore, if a non-'root admin' user tries to search for an 'Admin' role, this method will return null.
```
Comment (nitin-maharana) [inline api/src/main/java/org/apache/cloudstack/acl/RoleService.java]: @rafaelweingartner, Thanks for the fix. Would be more appropriate if we break this line into two.

Comment (rafaelweingartner) [reply api/src/main/java/org/apache/cloudstack/acl/RoleService.java]: Done.

**Comment by DaanHoogland** [comment]
looks good, I will run the tests though.
@blueorangutan package

File: server/src/test/java/org/apache/cloudstack/acl/RoleManagerImplTest.java [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+    public void beforeTest() {
+        Mockito.doReturn(accountMockId).when(accountMock).getId();
+        Mockito.doReturn(accountMock).when(roleManagerImpl).getCurrentAccount();
+
+        Mockito.doReturn(roleMockId).when(roleVoMock).getId();
+    }
+
+    @Test
+    public void findRoleTestIdNull() {
+        Role returedRole = roleManagerImpl.findRole(null);
```
Comment (nitin-maharana) [inline server/src/test/java/org/apache/cloudstack/acl/RoleManagerImplTest.java]: TYPO. I think you mean to be returnedRole( r -> n ). Next seven tests from here.

Comment (rafaelweingartner) [reply server/src/test/java/org/apache/cloudstack/acl/RoleManagerImplTest.java]: Fixed, thanks!

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by rafaelweingartner** [comment]
In the errors reported I found the following message `Exception: Host communication is not as expected: true. Instead, it's: false`. It looks like some action was taking longer than expected, and that is what causes the error. However, this PR does not change code that can have an effect on host status. Therefore, this is looking like an environment problem.

**Comment by rhtyd** [comment]
@rafaelweingartner do you want to consider this security/bug to be in 4.11? If yes, please edit the base branch of the PR and rebase branch, otherwise we can squash merge this PR.

**Pull request merged**